flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1214,34 +1214,67 @@ struct XEmboxOpConversion : public EmboxCommonConversion<fir::cg::XEmboxOp> {
     mlir::Value prevPtrOff = one;
     auto eleTy = boxTy.getEleTy();
     const auto rank = xbox.getRank();
-    for (unsigned di = 0, d = 0; di < rank; ++di) {
+    llvm::SmallVector<mlir::Value> gepArgs;
+    unsigned constRows = 0;
+    mlir::Value ptrOffset = zero;
+    if (auto memEleTy = fir::dyn_cast_ptrEleTy(xbox.memref().getType()))
+      if (auto seqTy = memEleTy.dyn_cast<fir::SequenceType>()) {
+        constRows = seqTy.getConstantRows();
+        auto seqEleTy = seqTy.getEleTy();
+        // Adjust the element scaling factor if the element is a dependent type.
+        if (fir::LLVMTypeConverter::dynamicallySized(seqEleTy)) {
+          if (fir::isa_char(seqEleTy)) {
+            assert(xbox.lenParams().size() == 1);
+            prevPtrOff = integerCast(loc, rewriter, i64Ty,
+                                     operands[xbox.lenParamOffset()]);
+          } else if (seqEleTy.isa<fir::RecordType>()) {
+            // prevPtrOff = ;
```

Comment:
Maybe put a more meaningful comment in place of this or just leave it out. Looks confusing by itself.